src/applications/lifecycles/bootstrap.js

```diff
@@ -1,21 +1,22 @@
-import { NOT_BOOTSTRAPPED, BOOTSTRAPPING, NOT_MOUNTED, SKIP_BECAUSE_BROKEN, getAppProps } from '../app.helpers.js';
+import { NOT_BOOTSTRAPPED, BOOTSTRAPPING, NOT_MOUNTED, SKIP_BECAUSE_BROKEN } from '../app.helpers.js';
 import { reasonableTime } from '../timeouts.js';
 import { handleAppError } from '../app-errors.js';
+import { getProps } from './prop.helpers.js'
 
-export async function toBootstrapPromise(app) {
-  if (app.status !== NOT_BOOTSTRAPPED) {
-    return app;
+export async function toBootstrapPromise(appOrParcel) {
+  if (appOrParcel.status !== NOT_BOOTSTRAPPED) {
+    return appOrParcel;
   }
 
-  app.status = BOOTSTRAPPING;
+  appOrParcel.status = BOOTSTRAPPING;
 
   try {
-    await reasonableTime(app.bootstrap(getAppProps(app)), `Bootstrapping app '${app.name}'`, app.timeouts.bootstrap);
-    app.status = NOT_MOUNTED;
+    await reasonableTime(appOrParcel.bootstrap(getProps(appOrParcel)), `Bootstrapping appOrParcel '${appOrParcel.name}'`, appOrParcel.timeouts.bootstrap);
```

Comment:
Are you also planning on converting this file to `Promises`?

Reply:
Yes converting to promises will be done all at once in one fell swoop